pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -566,6 +566,293 @@ func (gce *GCECloud) EnsureTCPLoadBalancerDeleted(name, region string) error {
 	return err
 }
 
+// UrlMap management
+
+// GetUrlMap returns the UrlMap by name.
+func (gce *GCECloud) GetUrlMap(name string) (*compute.UrlMap, error) {
+	return gce.service.UrlMaps.Get(gce.projectID, name).Do()
+}
+
+// CreateUrlMap creates an url map, using the given backend service as the default service.
+func (gce *GCECloud) CreateUrlMap(backend *compute.BackendService, name string) (*compute.UrlMap, error) {
```

Comment:
If we want this to be a generic method in the cloudprovider Interface (do we?), it shouldn't take/return compute pkg structs. But I suppose we can cross that bridge when we come to it.